src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -578,6 +590,199 @@ public static void registerTESRItemStack(Item item, int metadata, Class<? extend
+
+        List<BakedQuad> allquads = new ArrayList<>();
+
+        for (EnumFacing enumfacing : EnumFacing.VALUES)
+        {
+            allquads.addAll(model.getQuads(null, enumfacing, 0));
+        }
+
+        allquads.addAll(model.getQuads(null, null, 0));
+
+        // Current list of consecutive quads with the same lighting
+        List<BakedQuad> segment = new ArrayList<>();
+
+        // Lighting of the current segment
+        int segmentBlockLight = -1;
+        int segmentSkyLight = -1;
+        int segmentColorMultiplier = color;
+        
+        // Tint index cache to avoid unnecessary IItemColor lookups
+        int prevTintIndex = -1;
+
+        for (int i = 0; i < allquads.size(); i++) 
+        {
+            BakedQuad q = allquads.get(i);
+
+            // Lighting of the current quad
+            int bl = 0;
+            int sl = 0;
+
+            // Fail-fast on ITEM, as it cannot have light data
+            if (q.getFormat() != DefaultVertexFormats.ITEM) 
+            {
+                lightGatherer.setVertexFormat(q.getFormat());
+                if (lightGatherer.hasLighting())
+                {
+                    q.pipe(lightGatherer);
+                }
+                bl = lightGatherer.blockLight;
+                sl = lightGatherer.skyLight;
+            }
+            
+            int colorMultiplier = segmentColorMultiplier;
+            // If there is no color override, and this quad is tinted, we need to apply IItemColor
+            if (color == 0xFFFFFFFF && q.hasTintIndex())
+            {
+                int tintIndex = q.getTintIndex();
+                
+                if (prevTintIndex != tintIndex)
+                {
+                    colorMultiplier = getColorMultiplier(stack, tintIndex);
+                }
+                prevTintIndex = tintIndex;
+            }
+            else
+            {
+                colorMultiplier = color;
+                prevTintIndex = -1;
+            }
+
+            // If this is a new light value, draw the segment and flush it
+            if (segmentBlockLight != bl || segmentSkyLight != sl || segmentColorMultiplier != colorMultiplier) 
+            {
+                if (i > 0) // Make sure this isn't the first quad being processed 
+                {
+                    drawSegment(bufferbuilder, ri, segmentColorMultiplier, stack, segment, segmentBlockLight, segmentSkyLight, true);
+                }
+                segmentBlockLight = bl;
+                segmentSkyLight = sl;
+                segmentColorMultiplier = colorMultiplier;
+            }
+
+            segment.add(q);
+        }
+        
+        boolean lightDirty = segment.size() < allquads.size(); // Was at least one change in lighting for this model
+        drawSegment(bufferbuilder, ri, segmentColorMultiplier, stack, segment, segmentBlockLight, segmentSkyLight, lightDirty);
+
+        // Clean up render state if necessary
+        if (lightDirty)
```

Comment:
This is also incorrect, it **only** needs to check that the last segment was unlit.